bundle-core/src/main/java/net/discdd/utils/BundleUtils.java

```diff
@@ -56,12 +60,13 @@ public class BundleUtils {
     private static final String BUNDLE_EXTENSION = ".bundle";
     private static final String stringToMatch = "^[a-zA-Z0-9-_=]+$";
 
-    public static UncompressedBundle extractBundle(Bundle bundle, Path extractDirPath) {
+    public static UncompressedBundle extractBundle(Bundle bundle, Path extractDirPath) throws IOException {
         String bundleFileName = bundle.getSource().getName();
         logger.log(INFO, "Extracting bundle for bundle name: " + bundleFileName);
         Path extractedBundlePath = extractDirPath.resolve(bundleFileName);
         JarUtils.jarToDir(bundle.getSource().getAbsolutePath(), extractedBundlePath.toString());
-        File[] payloads = extractedBundlePath.resolve("payloads").toFile().listFiles();
+        Path path = extractDirPath.resolve(bundleFileName + "/payloads");
```

Comment:
i believe bundle.getSource() is a Path right? you can do bundle.getSource().resolve("payloads") (that helps you avoid assumptions about file separators)